File: examples-spark1/src/main/python/simple_sql_context.py
from mist.mist_job import *

class SimpleSQLContext(MistJob, WithSQLSupport):

    def execute(self, path):
        df = self.sql_context.read.json(path)
        df.printSchema()
        df.registerTempTable("people")

        df.show()
        df2 = self.sql_context.sql("SELECT AVG(age) AS avg_age FROM people")
        df2.show()

        result = df2.toJSON().first()

        return {"result": result}

File: examples-spark1/src/main/scala/SimpleSQLContext.scala
import io.hydrosphere.mist.api.{MistJob, SQLSupport}

object SimpleSQLContext extends MistJob with SQLSupport {
  /** Contains implementation of spark job with ordinary [[org.apache.spark.sql.SQLContext]]
    * Abstract method must be overridden
    *
    * @param file json file path
    * @return result of the job
    */
  def execute(file: String): Map[String, Any] = {
    val df = sqlContext.read.json(file)
    df.registerTempTable("people")
    Map("result" -> sqlContext.sql("SELECT AVG(age) AS avg_age FROM people").collect())
  }
}
